Source/Core/DolphinQt/Config/VerifyWidget.cpp

```diff
@@ -74,11 +74,17 @@ void VerifyWidget::CreateWidgets()
   std::tie(m_md5_checkbox, m_md5_line_edit) = AddHashLine(m_hash_layout, tr("MD5:"));
   std::tie(m_sha1_checkbox, m_sha1_line_edit) = AddHashLine(m_hash_layout, tr("SHA-1:"));
 
+  const auto default_to_calculate = DiscIO::VolumeVerifier::GetDefaultHashesToCalculate();
+  m_crc32_checkbox->setChecked(default_to_calculate.crc32);
+  m_md5_checkbox->setChecked(default_to_calculate.md5);
+  m_sha1_checkbox->setChecked(default_to_calculate.sha1);
+
   m_redump_layout = new QFormLayout;
   if (DiscIO::IsDisc(m_volume->GetVolumeType()))
   {
     std::tie(m_redump_checkbox, m_redump_line_edit) =
         AddHashLine(m_redump_layout, tr("Redump.org Status:"));
+    m_redump_checkbox->setChecked(CanVerifyRedump());
```

Comment:
I would really like Redump verification to always be enabled by default. Could we make it so that if the CPU supports fast CRC32 only, we still enable MD5 and/or SHA1?

Reply:
OK - I guess that logic should apply to both ui and dolphintool?

Reply:
Right now DolphinTool doesn't support Redump verification, right? But if we do add it, maybe it's best to just do it the simplest way possible, having Redump verification be directly controlled by a command-line flag. That is, unless you specify it, it doesn't run.

Reply:
oh, right

Reply:
Sorry, I think I somehow didn't fully realize that we were talking about the defaults for the hashes and not the default for whether to Redump verify. I think it would be best to use the same logic for DolphinTool, because even though it doesn't have Redump support, CRC32 is just really bad to use as a hash. (If you have NKit disc images and only hash them with CRC32, you'll get a match if you compare them with known good dumps!)

Reply:
CRC32 and MD5 (and soon, SHA1) are all useless if you want to use them to prevent purposeful tampering. It's just an artifact of working in an ecosystem revolving around dated game console.

Reply:
If your threat model is not an adversary who intentionally wants to trick you and has access to lots of computing power, MD5 and SHA1 are still much more useful than CRC32. I want to reiterate: Because of NKit, GC/Wii disc images with CRC32 collisions are now *common*. There's no software out there casually and instantly generating MD5 or SHA1 collisions for GC/Wii disc images like there is for CRC32.

Reply:
With that said, it would be nice to have a more secure hash available as well. But either way, I don't think we in any case should default to calculating only CRC32 because it would be faster.